bin/compile

```diff
@@ -1,2 +1,157 @@
 #!/usr/bin/env bash
 # Usage: bin/compile <build-dir> <cache-dir> <env-dir>
+# See: https://devcenter.heroku.com/articles/buildpack-api
+
+set -euo pipefail
+
+BUILD_DIR="${1}"
+CACHE_DIR="${2}"
+ENV_DIR="${3}"
+
+DOTNET_CNB_VERSION="0.1.7"
+
+BUILDPACK_DIR=$(cd "$(dirname "$(dirname "${BASH_SOURCE[0]}")")" && pwd)
+
+source "${BUILDPACK_DIR:?}/vendor/buildpack-stdlib_v8.sh"
+
+CNB_BUILD_DIR="/app/.heroku/cnb/dotnet"
+CNB_CACHE_DIR="${CACHE_DIR}/.heroku/cnb/dotnet"
+PROFILE_PATH="${BUILD_DIR}/.profile.d/dotnet.sh"
+
+initialize_environment() {
+	export_env "${ENV_DIR}" "." ""
+	mkdir -p "$(dirname "${PROFILE_PATH}")"
+	mkdir -p "${BUILD_DIR}/.heroku/cnb/dotnet"
+	mkdir -p "$(dirname "${CNB_BUILD_DIR}")"
+
+	# Create symlink from "/app/.heroku/cnb/dotnet" to the actual build directory.
+	# This allows the CNB to write path environment variables that work both at
+	# runtime and during the builds (particularly useful when a cached layer is
+	# restored, and the build dir has changed).
+	ln -nsf "${BUILD_DIR}/.heroku/cnb/dotnet" "${CNB_BUILD_DIR}"
+	touch "${CNB_BUILD_DIR}/buildpack_plan.toml"
+	mkdir -p "${CNB_BUILD_DIR}"/{layers,platform}
+
+	export CNB_BUILDPACK_DIR="${BUILDPACK_DIR}/cnb"
+	export CNB_TARGET_OS="linux"
+	local arch
+	arch=$(dpkg --print-architecture)
+	export CNB_TARGET_ARCH="${arch}"
+	export CNB_TARGET_DISTRO_NAME="ubuntu"
+	# shellcheck disable=SC2154 # TODO: Env var is referenced but not assigned.
+	export CNB_TARGET_DISTRO_VERSION="${STACK/heroku-/}.04"
+	export CNB_BUILD_DIR CNB_CACHE_DIR
```

Comment:
I think `CNB_BUILD_DIR` and `CNB_CACHE_DIR` might not be used be anything - and so maybe no need to export them?

`libcnb.rs` only uses these:

```
libcnb.rs $ rg '"CNB_.+"' -oI | sort -u
"CNB_BUILDPACK_DIR"
"CNB_TARGET_ARCH"
"CNB_TARGET_ARCH_VARIANT"
"CNB_TARGET_DISTRO_NAME"
"CNB_TARGET_DISTRO_VERSION"
"CNB_TARGET_OS"
```

And I don't see any usages in the .NET CNB itself:
https://github.com/search?q=repo%3Aheroku%2Fbuildpacks-dotnet+%2F%22CNB_.%2B%22%2F&type=code

If they are for future use, maybe worth adding a comment? :-)

Reply:
Good point -- I've removed both variables, and no longer assigning $2 to the (also) unused `CACHE_DIR`.